### PR TITLE
fix: Set `teamMemberEmail` server side for CRM RR Skip

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR finds the CRM contact owner server side and then passes the email to the client to be passed to the server when the booking is created.

This fixes relying on the booking form email field to find the contact owner which caused multiple re-renders.

Still valid
https://www.loom.com/share/29ae904eceed4aa5aced449a639d64a1?sid=3cad71bc-1def-49fc-84b2-e06fdf162b40

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Connect Salesforce and enable round robin skipping based on contact owner
- Go to the booking page and pass a Salesforce contact email as `email=...` in the URL query
	- Ensure that the contact owner is a part of your Salesforce instance
- The availability should only be queried for the contact owner

### CODE REVIEW

File: apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx [hunk 1]
```diff
@@ -81,6 +88,28 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
     } as const;
   }
 
+  let teamMemberEmail = null;
+
+  if (eventData.schedulingType === SchedulingType.ROUND_ROBIN && email) {
```
Comment (joeauyeung) [inline apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx:93]: Move finding the CRM contact owner from `useSchedule()` to `getServerSideProps()`

File: packages/app-store/_utils/CRMRoundRobinSkip.ts
Comment (joeauyeung) [inline packages/app-store/_utils/CRMRoundRobinSkip.ts:1]: Abstracted from `viewer/slots/util.ts`

File: packages/core/crmManager/crmManager.ts
Comment (joeauyeung) [inline packages/core/crmManager/crmManager.ts:1]: Pass the organizer email when creating a new contact so the organizer becomes the contact owner.

**Review by joeauyeung — COMMENTED** [review]
Simpler fix that utilizes the functionality we already have in `handleNewBooking`

**Comment by zomars** [comment]
Do we have an open issue for this?

**Comment by joeauyeung** [comment]
> Do we have an open issue for this?

@zomars this is the thread
https://app.campsite.co/cal/posts/07cxsuzbrrav

File: apps/web/test/lib/getSchedule.test.ts [hunk 2]
```diff
@@ -325,12 +320,10 @@ describe("getSchedule", () => {
           endTime: `${plus2DateString}T18:29:59.999Z`,
           timeZone: Timezones["+5:30"],
           isTeamEvent: true,
-          bookerEmail: "test@test.com",
+          teamMemberEmail: "example@example.com",
         },
       });
 
-      expect(scheduleFixedHostLead.teamMember).toBe("example@example.com");
```
Comment (zomars) [inline apps/web/test/lib/getSchedule.test.ts:332]: Are these expects non relevant anymore?

**Review by zomars — APPROVED** [review]

**Pull request merged**